core/src/plan/primary_key.rs

```diff
@@ -101,113 +195,398 @@ impl<'a> PrimaryKeyPlanner<'a> {
+                    } else {
+                        // Otherwise, we create a Partial primary key.
+                        PrimaryKey::new_partial(
+                            value.clone(),
+                            index,
+                            number_of_primary_key_columns,
+                            Expr::BinaryOp {
+                                left: Box::new(key.clone()),
+                                op: BinaryOperator::Eq,
+                                right: Box::new(value.clone()),
+                            },
+                        )
+                    }
+                })
         };
 
         match expr {
             Expr::BinaryOp {
-                left: key,
-                op: BinaryOperator::Eq,
-                right: value,
-            }
-            | Expr::BinaryOp {
-                left: value,
+                left,
                 op: BinaryOperator::Eq,
-                right: key,
-            } if check_primary_key(key.as_ref())
-                && check_evaluable(current_context.as_ref().map(Rc::clone), &key)
-                && check_evaluable(None, &value) =>
-            {
-                let index_item = IndexItem::PrimaryKey(*value);
-
-                PrimaryKey::Found {
-                    index_item,
-                    expr: None,
-                }
-            }
+                right,
+            } => get_primary_key(&left, &right)
+                .or_else(|| get_primary_key(&right, &left))
+                .unwrap_or_else(|| {
+                    PrimaryKey::NotFound(self.subquery_expr(
+                        Context::concat(Some(current_context.clone()), outer_context),
+                        Expr::BinaryOp {
+                            left,
+                            op: BinaryOperator::Eq,
+                            right,
+                        },
+                    ))
+                }),
             Expr::BinaryOp {
                 left,
                 op: BinaryOperator::And,
                 right,
             } => {
-                let primary_key = self.expr(
-                    outer_context.as_ref().map(Rc::clone),
-                    current_context.as_ref().map(Rc::clone),
-                    *left,
-                );
-
-                let left = match primary_key {
-                    PrimaryKey::Found { index_item, expr } => {
-                        let expr = match expr {
-                            Some(left) => Expr::BinaryOp {
-                                left: Box::new(left),
+                match (
+                    self.expr(
+                        outer_context.as_ref().map(Rc::clone),
+                        current_context,
+                        *left,
+                    ),
+                    self.expr(
+                        outer_context.as_ref().map(Rc::clone),
+                        current_context,
+                        *right.clone(),
+                    ),
```

Comment:
Before processing either the left or right variants, we need both of the results otherwise the subqueries are not handled properly (that was the issue beforehand). I am not sure I understood what you mean by "parallel". I just re-read the code and tried to see if I could find further simplifications, and I do not see any.

Reply:
ok, I will look deeper on it and leave comments. please let me have some more time on this.

Reply:
Cleaned up implementation more in commits [5f89a1d](https://github.com/gluesql/gluesql/pull/1515/commits/5f89a1d0b941dc62a24b895d78bebffb3623a32a) and 
f81d8ea